ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Workbenches.resource

```diff
@@ -485,4 +487,44 @@ Get Workbench Pod
     ${_}  ${cr_name}=    Get Openshift Notebook CR From Workbench
     ...    workbench_title=${workbench_title}  namespace=${ns_name}
     ${pod_info}=    Oc Get    kind=Pod  name=${cr_name}-0  api_version=v1  namespace=${ns_name}
-    RETURN    ${ns_name}    ${cr_name}    ${pod_info}
\ No newline at end of file
+    RETURN    ${ns_name}    ${cr_name}    ${pod_info}
+
+Edit Workbench
+    [Documentation]    Edits an existing workbench
+    [Arguments]    ${workbench_title}    ${pv_name}=${None}    ${workbench_title_update}=${None}    ${workbench_description}=${None}
+    ...     ${image_name}=${None}    ${version}=${None}   ${deployment_size}=${None}   ${storage}=${None}    ${pv_existent}=${None}
+    ...     ${envs}=${None}     ${dc_name}=${None}     ${aws_access_key}=${None}
+    ...     ${aws_secret_access}=${None}   ${aws_s3_endpoint}=${None}     ${aws_region}=${None}     ${gpus}=${NONE}
+    ...     ${data_connection}=${None}
+    Click Action From Actions Menu    item_title=${workbench_title}    item_type=workbench    action=Edit
+    Wait Until Page Contains Element    ${WORKBENCH_NAME_INPUT_XP}
+    IF    "${workbench_title_update}" != "${NONE}"   Input Text    ${WORKBENCH_NAME_INPUT_XP}    ${workbench_title_update}
+    IF    "${workbench_description}" != "${NONE}"    Input Text    ${WORKBENCH_DESCR_TXT_XP}    ${workbench_description}
+    IF    "${image_name}" != "${NONE}"    Select Workbench Jupyter Image    image_name=${image_name}    version=${version}
+    IF    "${deployment_size}" != "${NONE}"    Select Workbench Container Size    size_name=${deployment_size}
+    IF    "${gpus}" != "${NONE}"    Select Workbench Number Of GPUs    gpus=${gpus}
+    IF    "${envs}" != "${NONE}"
+        ${envs_copy}=    Copy List    ${envs}    deepcopy=${TRUE}
+        Add Environment Variables In Workbench    env_variables=${envs_copy}
+    END
+    IF    "${storage}" != "${NONE}"    Add Storage Details In Workbench    ${storage}  ${pv_existent}   ${pv_name}
+    ...    ${pv_description}  ${pv_size}
+    IF    "${data_connection}" == "new"
```

Comment:
are you going to implement the workflow for "existing"? 

Anyways, could we find a better name for this argument? Maybe `existing_data_connection=false`? otherwise `data_connection` seems like expecting the name or a pointer

Reply:
Moreover, what about implementing the same logic of `Add Storage Details In Workbench` instead of checking IF condition outside the `Add New Data Connection in Workbench` kw? it would be more consistent